app/src/main/java/org/p2p/wallet/updates/ConnectionStateProvider.kt

```diff
@@ -12,7 +13,7 @@ import kotlinx.coroutines.flow.onCompletion
 import kotlinx.coroutines.flow.onStart
 
 class ConnectionStateProvider(context: Context) {
-    private val connectivityManager = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
+    private val connectivityManager: ConnectivityManager = context.getSystemService()!!
```

Comment:
i think we should to avoid using !!

Reply:
agreed, do you have an alternative how to get System Services?